src/couch/src/couch_db.erl

```diff
@@ -1111,7 +1111,9 @@ prepare_doc_summaries(Db, BucketList) ->
                 nil
             end,
             SummaryChunk = couch_db_updater:make_doc_summary(Db, {Body, DiskAtts}),
-            Doc#doc{body = {summary, SummaryChunk, SizeInfo, AttsFd}}
+            Meta = Doc#doc.meta,
+            Doc#doc{body = {summary, SummaryChunk, SizeInfo, AttsFd},
+                meta = [{ejson_body, Body} | Meta]}
```

Comment:
You should just store the size of the body here rather than pass the whole body. By doing that you're effectively nullifying the summary optimization of only passing binaries in and out of couch_db_updater.